packages/core/hardhat.config.ts

```diff
@@ -242,9 +242,8 @@ const config: HardhatUserConfig = {
         network: 'polygonAmoy',
         chainId: 80002,
         urls: {
-          apiURL:
-            'https://www.oklink.com/api/v5/explorer/contract/verify-source-code-plugin/AMOY_TESTNET',
-          browserURL: 'https://www.oklink.com',
+          apiURL: 'https://api-amoy.polygonscan.com/api',
```

Comment:
We propably dont need custom chain for polygon scan, please check it

Reply:
we need it you can check `npx hardhat verify --list-networks ` it is not there. 
Also refer to : https://ethereum.stackexchange.com/questions/161654/unable-to-verify-contracts-on-polygon-amoy-testnet